packages/cmus/aaudio.patch

```diff
@@ -0,0 +1,994 @@
++	the system.
++
++	This option is only supported on Android 10+ (API 29) and requires
++	restarting cmus.
++
++dsp.aaudio.sharing_mode (shared) [shared, exclusive]
++	Request a mode for sharing the device. This is done on a best-effort
++	basis, and the requested sharing mode may not be available.
++
++	This option requires restarting cmus.
++
++dsp.aaudio.disable_spatialization (false)
++	If true, disables Android's built-in audio spatializer even if it is
++	enabled for the current audio output. Note that spatialization only
++	applies to audio with at least 4 channels. See
++	https://developer.android.com/media/grow/spatial-audio for more details.
++
++	This option is only supported on Android 12L+ (API 32) and requires
++	restarting cmus.
++
+ input.cdio.cddb_url
+ 	CDDB URL (default: freedb.freedb.org:8880). Uses HTTP if prefixed with
+ 	"http://" (e.g.: http://freedb.musicbrainz.org:80/~cddb/cddb.cgi). Set
+diff --git a/Makefile b/Makefile
+index 4944182..8662645 100644
+--- a/Makefile
++++ b/Makefile
+@@ -180,6 +180,7 @@ ao-objs			:= op/ao.lo
+ coreaudio-objs		:= op/coreaudio.lo
+ waveout-objs		:= op/waveout.lo
+ roar-objs               := op/roar.lo
++aaudio-objs		:= op/aaudio.lo
+ 
+ op-$(CONFIG_PULSE)	+= op/pulse.so
+ op-$(CONFIG_ALSA)	+= op/alsa.so
+@@ -192,6 +193,7 @@ op-$(CONFIG_COREAUDIO)	+= op/coreaudio.so
+ op-$(CONFIG_AO)		+= op/ao.so
+ op-$(CONFIG_WAVEOUT)	+= op/waveout.so
+ op-$(CONFIG_ROAR)       += op/roar.so
++op-$(CONFIG_AAUDIO)	+= op/aaudio.so
+ 
+ $(pulse-objs): CFLAGS		+= $(PULSE_CFLAGS)
+ $(alsa-objs): CFLAGS		+= $(ALSA_CFLAGS)
+@@ -204,6 +206,7 @@ $(ao-objs):   CFLAGS		+= $(AO_CFLAGS)
+ $(coreaudio-objs): CFLAGS	+= $(COREAUDIO_CFLAGS)
+ $(waveout-objs): CFLAGS 	+= $(WAVEOUT_CFLAGS)
+ $(roar-objs): CFLAGS		+= $(ROAR_CFLAGS)
++$(aaudio-objs): CFLAGS		+= $(AAUDIO_CFLAGS)
+ 
+ op/pulse.so: $(pulse-objs) $(libcmus-y)
+ 	$(call cmd,ld_dl,$(PULSE_LIBS))
+@@ -237,6 +240,9 @@ op/waveout.so: $(waveout-objs) $(libcmus-y)
+ 
+ op/roar.so: $(roar-objs) $(libcmus-y)
+ 	$(call cmd,ld_dl,$(ROAR_LIBS))
++
++op/aaudio.so: $(aaudio-objs) $(libcmus-y)
++	$(call cmd,ld_dl,$(AAUDIO_LIBS))
+ # }}}
+ 
+ # man {{{
+diff --git a/configure b/configure
+index fb750ad..c1a4216 100755
+--- a/configure
++++ b/configure
+@@ -498,6 +498,37 @@ check_ffmpeg()
+ 	return 1
+ }
+ 
++aaudio_code="
++#include <aaudio/AAudio.h>
++int main() {
++	if (__builtin_available(android 26, *)) {
++		AAudioStreamBuilder *bld;
++		if (AAudio_createStreamBuilder(&bld))
++			return 1;
++		if (AAudioStreamBuilder_delete(bld))
++			return 1;
++	}
++	return 0;
++}
++"
++
++check_aaudio()
++{
++	if check_library AAUDIO "" "-laaudio"
++	then
++		AAUDIO_CFLAGS="$AAUDIO_CFLAGS -D__ANDROID_UNAVAILABLE_SYMBOLS_ARE_WEAK__ -Werror=unguarded-availability"
++
++		msg_checking "for aaudio support"
++		if try_compile_link "$aaudio_code" $AAUDIO_CFLAGS $AAUDIO_LIBS
++		then
++			msg_result yes
++			return 0
++		fi
++		msg_result no
++	fi
++	return 1
++}
```

Comment:
I will need to change this logic since it isn't adding the linker flags when compiled with the package build.sh, probably due to the weak linking causing it to succeed without the `-laaudio`.

Reply:
We could always add `LDFLAGS+=' -laaudio'` in the preconfigure step.

Reply:
Only the output plugin should be linked to audio, otherwise, cmus won't work on versions without it anymore.

Reply:
Ah yep that makes sense.

Reply:
I think I've fixed the issue; it was due to weak linking and the toolchain config.